tests/utils/test_schema_conversion.py

```
@@ -0,0 +1,897 @@
+        """A doubling ref chain (billion laughs) is stopped by max_nodes."""
+        defs = {
+            f"L{i}": _object(
+                {
+                    "a": {"$ref": f"#/$defs/L{i + 1}"},
+                    "b": {"$ref": f"#/$defs/L{i + 1}"},
+                }
+            )
+            for i in range(10)
+        }
+        defs["L10"] = {"type": "string"}
+        with pytest.raises(ValueError, match="maximum of .* nodes"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)